tools/serverpod_cli/test/generator/open_api/generate_openapi_test.dart

```diff
@@ -0,0 +1,246 @@
+import 'dart:convert';
+
+import 'package:path/path.dart' as path;
+import 'package:serverpod_cli/src/generator/open_api/open_api_objects.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+import 'test_data_factory.dart';
+
```

Comment:
Suggestion: As mentioned in other comments, focus on testing on this abstraction layer, and be detailed in the expects.

This means a lot of different combinations of test data. Test endpoints with different return types, different input types, protocol files with different scopes, protocol files that are used / not used, etc, etc.